AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -703,6 +704,69 @@ - (instancetype)styledWithBlock:(void (^)(ASLayoutElementStyle *style))styleBloc
 
 #pragma mark - Layout
 
+- (void)invalidateSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  
+  __instanceLock__.lock();
+
+  // Mark the node for layout in the next layout pass
+  [self setNeedsLayout];
+  
+  ASDisplayNode *supernode = _supernode;
+  if (supernode) {
+    __instanceLock__.unlock();
+    // Cause supernode's layout to be invalidated
+    // We need to release the lock to prevent a deadlock
+    [supernode invalidateSize];
+    return;
+  }
+  
+  // We are the root node and need to re-flow the layout; one of our children requested to have its size re-set.
+  CGSize oldSize = self.bounds.size;
+
+  // Figure out constrainedSize to use
+  ASSizeRange constrainedSize = ASSizeRangeMake(self.bounds.size);
+  if (_pendingDisplayNodeLayout != nullptr) {
+    constrainedSize = _pendingDisplayNodeLayout->constrainedSize;
+  } else if (_calculatedDisplayNodeLayout->layout != nil) {
+    constrainedSize = _calculatedDisplayNodeLayout->constrainedSize;
+  }
+
+  // Check if the returned layout has a different size as the current bounds
+  ASLayout *layout = [self layoutThatFits:constrainedSize];
+  if (CGSizeEqualToSize(oldSize, layout.size) == NO) {
+    // If the size of the layout changes inform our container (e.g ASTableView, ASCollectionView, ASViewController, ...)
+    // that we need it to change our bounds size.
+    [self displayNodeDidInvalidateSizeNewSize:layout.size];
```

Comment:
Is it possible to just use -calculatedLayoutDidChange here?  Would providing internal access (for ASCellNode's use case) to _pendingLayout, so .size can be accessed, be enough?  Not a big deal as long as the method moves to FrameworkPrivate.

If we did use another method without an argument, then the size != size check would have to be in ASCellNode before calling its table / collection.